pkg/cloudprovider/provider/digitalocean/provider.go

```diff
@@ -268,19 +266,20 @@ func (p *provider) Create(machine *v1alpha1.Machine, userdata string) (instance.
 	ctx := context.TODO()
 	client := getClient(c.Token)
 
-	// TODO(p0lyn0mial): an ssh-key should be created only if the CoreOS was requested
-	// TODO(p0lyn0mial): an ssh-key must be temporal - deleted after an instance has been created.
-	if p.privateKey == nil {
-		var err error
-		p.privateKey, err = machinessh.NewPrivateKey("machine-controller")
-		if err != nil {
-			return nil, err
-		}
+	tmpSSHKeyPair, err := machinessh.NewPrivateKey("machine-controller-do-tmp-key")
```

Comment:
I think the whole `machinessh` isn't necessary anymore.

Reply:
movie it to `pkg/cloudprovider/provider/digitalocean/provider.go` ?

Reply:
Yeah, probably. Though it contains a wrapper around the actual ssh key, which we dont need anymore.

The only code we need is 
```go
	priv, err := rsa.GenerateKey(rand.Reader, privateKeyBitSize)
	if err != nil {
		return nil, fmt.Errorf("failed to create private key: %v", err)
	}
	publicKey := priv.PublicKey
```

Reply:
ok.